src/workerd/io/trace.c++

```diff
@@ -621,6 +624,7 @@ void WorkerTracer::addDiagnosticChannelEvent(kj::Date timestamp,
                    message.size();
   if (newSize > MAX_TRACE_BYTES) {
```

Comment:
I think there's also a less-likely `if (newSize > MAX_TRACE_BYTES)` case in the fetch event, to cover things like large header or cfJson values.